data_structures/binary_tree/is_bst.py

```diff
@@ -0,0 +1,117 @@
+"""
+Author  : Alexander Pantyukhin
+Date    : November 2, 2022
+
+Task:
+Given the root of a binary tree, determine if it is a valid binary search
+tree (BST).
+
+A valid BST is defined as follows:
+
+- The left subtree of a node contains only nodes with keys less than the node's key.
+- The right subtree of a node contains only nodes with keys greater than the node's key.
+- Both the left and right subtrees must also be binary search trees.
+
+Implementation notes:
+Depth-first search approach.
+
+leetcode: https://leetcode.com/problems/validate-binary-search-tree/
+
+Let n is the number of nodes in tree
+Runtime: O(n)
+Space: O(1)
+"""
+
+from __future__ import annotations
+
+
+class TreeNode:
+    def __init__(
+        self,
+        data: float = 0,
+        left: TreeNode | None = None,
+        right: TreeNode | None = None,
+    ) -> None:
+        self.data = data
+        self.left = left
+        self.right = right
+
+
+def is_bst(root: TreeNode | None) -> bool:
```

Comment:
For readability, we try to expand acronyms so `is_binary_search_tree()` or maybe even better...
```suggestion
def is_balanced(root: TreeNode | None) -> bool:
```

Reply:
I applied is_binary_search_tree. I think it's more close that the func does. If I correctly understand: balanced - heights keft right is close to each other